man/coap_block.txt.in

```diff
@@ -0,0 +1,457 @@
+series of packet interchanges.  Furthermore, if COAP_BLOCK_SINGLE_BODY is set,
+then the PDU that presents the entire body will have any BLOCKx option removed.
+
+*NOTE:* COAP_BLOCK_USE_LIBCOAP must be set if libcoap is to do all the
+block tracking and requesting, otherwise the application will have to do all
+of this work (the default if *coap_context_set_block_mode*() is not called).
+
+[source, c]
+----
+/**
+ * Callback handler for de-allocating the data based on @p app_ptr provided to
+ * coap_add_data_large_*() functions following transmission of the supplied
+ * data.
+ *
+ * @param session The session that this data is associated with
+ * @param app_ptr The application provided pointer to the
+ *                coap_add_data_large_*() functions
+ */
+typedef void (*coap_release_large_data_t)(struct coap_session_t *session,
+                                          void *app_ptr);
+----
+
+The *coap_add_data_large_request*() function is similar to *coap_add_data*(),
+but supports the transmission of data that has a body size that is potentially
+larger than can be fitted into a single client request PDU. The specified
+payload _data_ of length _length_ is associated with the _session_ with the
+first block of data added to the PDU _pdu_ along with the appropriate CoAP
+options such as BLOCK1, and SIZE1 if the data does not fit into
+a single PDU.  When the block has been acknowledged by the peer, the library
+will then send the next block of data until all the data has been transmitted.
+This function must only be called once per _pdu_.  When the final block is
+transmitted, the callback function _release_func_ (if not NULL) with the user
+defined parameter of _app_ptr_ is called so that the data can be released.
+
+The *coap_add_data_large_response*() function is responsible for handling
+the server's large responses to requests. *coap_add_data_large_response*()
+should be used as a direct replacement for *coap_add_data*() if it is possible
+that the _length_ of _data_ will not fit into a single server's response pdu.
+This function adds in the initial part of the payload _data_ of length
+_length_ to the PDU _pdu_. _release_func_ (if not NULL) and _app_ptr_ are
+used for releasing the data when the body transfer is complete.  It also adds
+in the appropriate CoAP options such as BLOCK2, SIZE2 and ETAG to handle
+Block-Wise transfer if the data does not fit into a single PDU.
+_resource_, _query_, _session_, _request_, _response_ and _token_ are the same
+parameters as in the called resource handler that invokes
+*coap_add_data_large_response*(). If _etag_ is 0, then a unique ETag value will
+be generated, else is the ETag value to use.
+The _media_type_ is for the format of the _data_ and _maxage_ defines the
+lifetime of the response.  If _maxage_ is set to -1,  then the MAXAGE option
+does not get included.  This function must only be called once per _pdu_.
```

Comment:
I suggest adding "(which indicates the default value of 60 seconds according to RFC 7252)" after "does not get included".

Reply:
Text added